### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by petdance** [htacg/tidy-html5#59]
Since we're starting a new project, let's take the time up front to clean up some of the artifacts from the old tidy:

* Standardize tabs/spaces and indenting, including adding vim mode lines (and emacs if anyone wants them)
* Clean all spurious whitespace so that we don't have false diffs down the road.
* Remove all CVS cruft.  These have no value to us:

```
   CVS Info:
     $Author: arnaud02 $
     $Date: 2007/02/11 09:45:08 $
     $Revision: 1.9 $
```

**Comment by balthisar** [comment]
I'm in favor of biting the bullet and fixing this as our last operation before merging into Master. Future PRs will be affected, so a code freeze is a pretty going point to do this, too.

**Comment by geoffmcl** [comment]
@petdance, @balthisar, did a quick analysis of the 52 source files, 50,352 lines, approx. 2.7MB... trying to determine what, if anything, needs 'fixing' in this regard.

Some 1.4% of the lines, 722 lines, in 34 files, have 1 or more trailing spaces. Does not seem like a giant problem ;=))

Regarding tabs, only 4 files contain tabs  

 1. tidylib.c - 2144 lines, 4 w/tab, 4 tabs
 2. sprtf.c - 440 lines, 45 w/tab, 142 tabs
 3. parser.c - 4898 lines, 124 w/tab, 341 tabs
 4. tidy-int.h - 155 lines, 6 w/tab, 19 tabs

The sprtf.c is my windows ONLY debug addition, so this should not be a problem. This file is not included/compiled in the release build, and is only ever 'viewed' in my MSVC IDE which is set to display tabs as 4 spaces, so it still all lines up neatly...

parser.c is because I (or someone?) imported some code from another repo that used tabs. I did remove some, but a few remain... At 150K bytes it is the 2nd largest source file in tidy.

The largest is attrdict.c, 13,827 lines, 1.5MB, with no tabs, and only 131 lines with 1 or more trailing spaces.

But with a grand total of just 506 tabs, 0.01% of the total bytes, again it does not seem a problem.

As for relining, and indenting, having been through and through most of the source over the last few months, I have NOT seen anything noticeably bad...

In fact I **LOVE** that doing a big diff between cvs/sf tidy and current tidy5, there are still thousands of lines that are totally **UNCHANGED**. This is particular good for doing code reviews, comparison, etc...

So in summary, maybe I missed something, but I would not recommend this be done at this time.

But on the other hand, if someone wants to do this, as a PR, I would certainly look at what is about to be changed. And this is certainly the time to do it before releasing 5.0.0. But again I hope there are still thousand of lines left unchanged as mentioned above.

A feature request that I would like to pass to git is to support a -w option on the diff, but given that git is totally line oriented, I guess this would not be easily accepted. My GUI side-by-side file comparer is WinMerge, and it thankfully chooses by default to ignore pure space changes.

Anyway, just my 2 cent ;=))

**Comment by geoffmcl** [comment]
No further comments for over a month, so closing...

Feel free to re-open, or raise another issue... thanks.
